### PR TITLE
feat: add support for cp4d auth

### PR DESCRIPTION
**Pull request by rfazeli**
Add support for cp4d authentication

### CODE REVIEW

File: assistant_skill_analysis/utils/skills_util.py [hunk 1]
```diff
@@ -140,7 +142,12 @@ def retrieve_conversation(
     if iam_apikey:
         authenticator = IAMAuthenticator(apikey=iam_apikey, url=authenticator_url)
     elif username and password:
-        authenticator = BasicAuthenticator(username=username, password=password)
+        if cp4d_auth:
```
Comment (haodeqi) [inline assistant_skill_analysis/utils/skills_util.py:147]: can we also include an additional option to use Bearer Authenticator? https://github.com/IBM/python-sdk-core/blob/main/Authentication.md#bearer-token-authentication

**Review by haodeqi — CHANGES_REQUESTED** [review]
Hi Reza, thanks for the update. I have a inline comment for you.

File: assistant_skill_analysis/utils/skills_util.py (hunk 1 above)
Comment (rfazeli) [reply assistant_skill_analysis/utils/skills_util.py:147]: Good point! Do we want to update the notebooks to show the bearer token use case?

Comment (haodeqi) [reply assistant_skill_analysis/utils/skills_util.py:147]: that will be good. Can you add it to the authentication section and comment it out with some instructions of how to use that?

Comment (rfazeli) [reply assistant_skill_analysis/utils/skills_util.py:147]: Got it! Notebooks should be good now.

**Comment by rfazeli** [comment]
@haodeqi some unit tests seem to be failing but I don't think they're related to the changes here (at least based on the failures I see when running pytest locally).

File: assistant_skill_analysis/utils/skills_util.py (hunk 1 above)
Comment (sodoherty-ai) [reply assistant_skill_analysis/utils/skills_util.py:147]: You need to add the BearerToken and CP4D authenticators.

**Comment by sodoherty-ai** [comment]
The bearer token line should be: 

`conversation = skills_util.retrieve_conversation(bearer_token=BEARER_TOKEN, url=SERVICE_URL)`

I would also recommend to move `BEARER_TOKEN =` to the line just above this line.

File: action_skill_analysis_cp4d.ipynb [hunk 2]
```diff
@@ -0,0 +1,995 @@
...
+    "password = '###'\n",
+    "SERVICE_URL = 'service_url'\n",
+    "CP4D_URL = 'cp4d_url'\n",
+    "BEARER_TOKEN = 'bearer_token' # this is optional\n",
+    "\n",
+    "# path to downloaded action skill workspace json\n",
+    "\n",
+    "ACTION_SKILL_FILENAME = '###'\n",
+    "\n",
+    "# # choose a datacenter to use\n",
```
Comment (haodeqi) [inline action_skill_analysis_cp4d.ipynb]: let's remove this block since it isnt needed for cp4d

File: action_skill_analysis_cp4d.ipynb [hunk 3]
```diff
@@ -0,0 +1,995 @@
...
+    "\n",
+    "lang_util = lang_utils.LanguageUtility(LANGUAGE_CODE)\n",
+    "\n",
+    "_, _, ASSISTANT_ID = skills_util.input_credentials(input_apikey=False,input_workspace_id=False,input_assistant_id=True)\n",
+    "\n",
+    "username = 'apikey'\n",
+    "password = '###'\n",
+    "SERVICE_URL = 'service_url'\n",
+    "CP4D_URL = 'cp4d_url'\n",
+    "BEARER_TOKEN = 'bearer_token' # this is optional\n",
```
Comment (haodeqi) [inline action_skill_analysis_cp4d.ipynb]: let's move this right above the line of `skills_util.retrieve_conversation(bearer_token=BEARER_TOKEN, url=SERVICE_URL)`

File: action_skill_analysis_cp4d.ipynb [hunk 4]
```diff
@@ -0,0 +1,995 @@
...
+    "                                             password=password,                                                       \n",
+    "                                             authenticator_url=CP4D_URL,\n",
+    "                                             url=SERVICE_URL,\n",
+    "                                             api_version=API_VERSION,\n",
+    "                                             sdk_version=\"V2\", # DO NOT CHANGE THIS ARG\n",
+    "                                             cp4d_auth=True,\n",
+    "                                             )\n",
+    "\n",
+    "# if you have a BEARER_TOKEN, you can uncomment the code below and directly use your token\n",
+    "# conversation = skills_util.retrieve_conversation(bearer_token=BEARER_TOKEN)\n",
```
Comment (haodeqi) [inline action_skill_analysis_cp4d.ipynb]: it should be `skills_util.retrieve_conversation(bearer_token=BEARER_TOKEN, url=SERVICE_URL)`

File: dialog_skill_analysis_cp4d.ipynb [hunk 5]
```diff
@@ -0,0 +1,1116 @@
...
+    "\n",
+    "conversation = skills_util.retrieve_conversation(username=username,\n",
+    "                                             password=password,                                                       \n",
+    "                                             authenticator_url=CP4D_URL,\n",
+    "                                             url=SERVICE_URL,\n",
+    "                                             api_version=API_VERSION,\n",
+    "                                             cp4d_auth=True,\n",
+    "                                             )\n",
+    "\n",
+    "# if you have a BEARER_TOKEN, you can uncomment the code below and directly use your token\n",
```
Comment (haodeqi) [inline dialog_skill_analysis_cp4d.ipynb:167]: same here

**Review by haodeqi — CHANGES_REQUESTED** [review]
Hi Reza, thanks for the update. I have a few more comments for you. Additionally, can you change the stopwords to a list, the more recent version of sicikit-learn is stricter regarding input types. https://github.com/watson-developer-cloud/assistant-skill-analysis/blob/master/assistant_skill_analysis/utils/lang_utils.py#L111

**Comment by rfazeli** [comment]
Thanks for your feedback @haodeqi @sodoherty-ai! I've addressed the comments

**Review by haodeqi — APPROVED** [review]
Looks good to me.

**Pull request merged**